packages/esm-patient-registration-app/src/patient-registration/field/id/id-field.test.tsx

```diff
@@ -0,0 +1,164 @@
+    defaultPatientIdentifierTypes: ['OpenMRS ID'],
+  }));
+
+  it('should render loading skeleton when identifier types are loading', () => {
+    render(
+      <ResourcesContext.Provider value={[]}>
+        <Formik initialValues={{}} onSubmit={null}>
+          <Form>
+            <PatientRegistrationContext.Provider
+              value={{
+                setFieldValue: jest.fn(),
+                initialFormValues: { identifiers: { openmrsID } },
+                setInitialFormValues: jest.fn(),
+                values: {
+                  identifiers: { openmrsID },
+                },
+              }}>
+              <Identifiers />
+            </PatientRegistrationContext.Provider>
+          </Form>
+        </Formik>
+      </ResourcesContext.Provider>,
+    );
+    expect(screen.getByTestId('loading-skeleton')).toBeInTheDocument();
+  });
+
+  it('should render identifier inputs when identifier types are loaded', () => {
+    render(
+      <ResourcesContext.Provider value={mockResourcesContextValue}>
+        <Formik initialValues={{}} onSubmit={null}>
+          <Form>
+            <PatientRegistrationContext.Provider
+              value={{
+                setFieldValue: jest.fn(),
+                initialFormValues: { identifiers: { openmrsID } },
+                setInitialFormValues: jest.fn(),
+                values: {
+                  identifiers: { openmrsID },
+                },
+              }}>
+              <Identifiers />
+            </PatientRegistrationContext.Provider>
+          </Form>
+        </Formik>
+      </ResourcesContext.Provider>,
+    );
+
+    expect(screen.getByText('Identifiers')).toBeInTheDocument();
+    const configureButton = screen.getByRole('button', { name: 'Configure' });
+    expect(configureButton).toBeInTheDocument();
+    expect(configureButton).toBeEnabled();
+  });
+
+  it('should open identifier selection overlay when "Configure" button is clicked', () => {
+    render(
+      <ResourcesContext.Provider value={mockResourcesContextValue}>
+        <Formik initialValues={{}} onSubmit={null}>
+          <Form>
+            <PatientRegistrationContext.Provider
+              value={{
+                setFieldValue: jest.fn(),
+                initialFormValues: { identifiers: { openmrsID } },
+                setInitialFormValues: jest.fn(),
+                values: {
+                  identifiers: { openmrsID },
+                },
+              }}>
+              <Identifiers />
+            </PatientRegistrationContext.Provider>
+          </Form>
+        </Formik>
+      </ResourcesContext.Provider>,
+    );
+
+    const configureButton = screen.getByRole('button', { name: 'Configure' });
+    fireEvent.click(configureButton);
+
+    expect(screen.getByRole('button', { name: 'Close overlay' })).toBeInTheDocument();
+  });
+
+  it('should close identifier selection overlay when "close overlay" button is clicked', () => {
```

Comment:
I think this test is not related to the id-field component. It is something `ui-components/overlay` is responsible for.

Reply:
Yes @jayasanka-sack you are right. This is coming from the overlay component. `id-field.component.tsx` is using `identifier-selection-overlay.tsx` as child component which is wrapper implementation of `ui-components/overlay`. But there is one issue when I will test this behaviour in test for `ui-components/overlay` then the ui changes will not happen. for example the close button will not close the overlay component it will just call the onClose function. So in this case I am not sure what to do ? If I am testing this in parent component the ui change is happening. So where should I place this test?
cc/ @Piumal1999 @anjula-sack